include/swift/AST/Decl.h

```diff
@@ -2265,6 +2265,13 @@ class ValueDecl : public Decl {
     return result;
   }
 
+  /// SE-0025 corner case -- we allow members to have more visible access levels
+  /// than their enclosing scope. Normally, we won't find such members if the
+  /// containing scope is not really visible. However, if the containing scope
+  /// is a protocol extension, they can show up here, so we have to filter
+  /// them out directly.
+  AccessLevel adjustAccessLevelForProtocolExtension(AccessLevel access) const;
```

Comment:
Nitpick: this should be `static`.

Reply:
It calls getDeclContext() on this

Reply:
Uh…that seems fishy, then. Why isn't it only called when needed?